fluid/voice.cpp

```diff
@@ -1924,5 +1958,30 @@ void Voice::effects(int startBufIdx, int count, float* out, float* reverb, float
                   }
             }
       }
-}
 
+      /** legato update functions --------------------------------------------------*/
+      /* Updates voice portamento parameters
+       *
+       * @fromkey the beginning pitch of portamento.
+       * @tokey the ending pitch of portamento.
+       *
+       * The function calculates pitch offset and increment, then these parameters
+       * are send to the dsp.
+      */
+      void Voice::updatePortamento(int fromkey, int tokey) {
+            /* calculates pitch offset */
+            float PitchBeg = gen[GEN_SCALETUNE].val
+                  * (fromkey - root_pitch / 100.0f) + root_pitch;
+            float PitchEnd = gen[GEN_SCALETUNE].val
+                  * (tokey - root_pitch / 100.0f) + root_pitch;
+            float pitchoffset = PitchBeg - PitchEnd;
+            setPortamento(samplesToNextTurningPoint(channel), pitchoffset);
+            }
+
+      void Voice::setPortamento(unsigned int countinc, float pitchoffset) {
+            if (countinc) {
+                  this->pitchoffset += pitchoffset;
+                  this->pitchinc = pitchoffset/(_fluid->sampleRate()*(0.001f*countinc));//countinc;
+                  }
+            }
```

Comment:
This section seems indented one level too much

Reply:
Resolved